### PR TITLE
feat: authentication

### PR DESCRIPTION
**Pull request by AhmedAlzubaidi**

### CODE REVIEW

File: ClassiVault.Api.csproj [hunk 1]
```diff
@@ -8,6 +8,8 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Identity.EntityFrameworkCore" Version="7.0.13" />
```
Comment (AbdulmajeedAlOmari) [inline ClassiVault.Api.csproj]: Hey Ahmad, there is a newer version for entity framework: [8.0.0-rc.2.23480.2](https://www.nuget.org/packages/Microsoft.AspNetCore.Identity.EntityFrameworkCore/8.0.0-rc.2.23480.2)

File: ClassiVault.Api.csproj [hunk 2]
```diff
@@ -8,6 +8,8 @@
...
+    <PackageReference Include="Microsoft.EntityFrameworkCore.InMemory" Version="7.0.13" />
```
Comment (AbdulmajeedAlOmari) [inline ClassiVault.Api.csproj]: There is a newer version: [8.0.0-rc.2.23480.1](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.InMemory/8.0.0-rc.2.23480.1)

File: Models/AppDbContext.cs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+using Microsoft.AspNetCore.Identity.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore;
+
+namespace ClassiVault.Api.Models
+{
+    class AppDbContext(DbContextOptions<AppDbContext> options) : IdentityDbContext<User>(options) {
```
Comment (AbdulmajeedAlOmari) [inline Models/AppDbContext.cs:6]: Can we reuse this AppDbContext for the other tables? Password encryptions tables... etc

**Review by AbdulmajeedAlOmari — CHANGES_REQUESTED** [review]
That's awesome work @AhmedAlzubaidi! Now I'm worried that my code doesn't match the quality of yours 😢

File: Models/AppDbContext.cs (hunk 3 above)
Comment (AhmedAlzubaidi) [reply Models/AppDbContext.cs:6]: Yes according to shaikh ChatGPT we could

File: ClassiVault.Api.csproj (hunk 1 above)
Comment (AhmedAlzubaidi) [reply ClassiVault.Api.csproj]: Updated.

File: ClassiVault.Api.csproj (hunk 2 above)
Comment (AhmedAlzubaidi) [reply ClassiVault.Api.csproj]: Updated.

**Pull request merged**